cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -90,17 +90,6 @@ func NewServiceAccountSigningKey() (*rsa.PrivateKey, error) {
 	return saSigningKey, nil
 }
 
-// NewCACertAndKey will generate a self signed CA.
-func NewCACertAndKey(certSpec *certutil.Config) (*x509.Certificate, *rsa.PrivateKey, error) {
-
-	caCert, caKey, err := pkiutil.NewCertificateAuthority(certSpec)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "failure while generating CA certificate and key")
```

Comment:
IMO, it's provider a better error message, using only `pkiutil.NewCertificateAuthority` will return `unable to create private key` or `unable to create self-signed certificate`.
I'm not against the change, but will leave the lgtm to other reviews.

Reply:
If the sole purpose of this function is to give a better error message then the messages returned by `pkiutil.NewCertificateAuthority` could be made clearer, e.g. `unable to create private key while generating CA certificate` and `unable to create self-signed CA certificate`. What do you think?

Also additional error wrapping goes against the rationale explained in https://github.com/kubernetes/kubeadm/issues/1266

Reply:
You both have a point here. The func was introduced in its form by Fabrizio, so it's probably best to let him decide on this one.

/assign @fabriziopandini